core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -104,10 +115,20 @@ SdkPojo unmarshall(XmlUnmarshallerContext context, SdkPojo sdkPojo, XmlElement r
                                                root.getElementsByName(field.unmarshallLocationName());
 
                     if (!CollectionUtils.isNullOrEmpty(element)) {
-                        boolean isFieldBlobTypePayload = payloadMemberAsBlobType.isPresent() &&
-                                                         payloadMemberAsBlobType.get().equals(field);
-                        if (isFieldBlobTypePayload) {
+                        if (isExplicitPayloadMember(field) && field.marshallingType() == MarshallingType.SDK_BYTES) {
                             field.set(sdkPojo, SdkBytes.fromInputStream(context.response().content().get()));
+                        } else if (isExplicitPayloadMember(field) && field.marshallingType() == MarshallingType.STRING) {
+                            SdkBytes sdkBytes = SdkBytes.fromInputStream(context.response().content().get());
+                            String stringPayload = sdkBytes.asUtf8String();
+                            if (isXmlString(stringPayload)) {
+                                InputStream inputStream = new ByteArrayInputStream(sdkBytes.asByteArray());
+                                XmlElement document = XmlDomParser.parse(inputStream);
+                                Object unmarshalled = unmarshaller.unmarshall(context, singletonList(document),
+                                                                              (SdkField<Object>) field);
+                                field.set(sdkPojo, unmarshalled);
```

Comment:
When can this happen? It's marked as an explicit string payload, but it needs to be parsed like XML?

Reply:
Yeah I initially did not include this, but these tests broke
- https://github.com/aws/aws-sdk-java-v2/blob/master/test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/RestXmlEventStreamProtocolTest.java#L117
- https://github.com/aws/aws-sdk-java-v2/blob/master/test/protocol-tests/src/test/java/software/amazon/awssdk/protocol/tests/RestXmlEventStreamProtocolTest.java#L133

I would think that if someone marks as explicit String payload that they wouldn't pass in XML formatted. Any chance we break existing customers?Would it be fine to just update the assertion in these tests to the full XML payload?

Reply:
Checking with smithy on expected behavior, pending response.
Looks like we need to at least handle the special case for S3's `GetBucketPolicy`, which we have a [custom interceptor](https://github.com/aws/aws-sdk-java-v2/blob/master/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/handlers/GetBucketPolicyInterceptor.java#L40) for that wraps the payload in XML.

Reply:
We should [umarshall the response to exclude the tags](https://github.com/smithy-lang/smithy/blob/eedc319c3cbfda80921a270d473ba97072287e83/smithy-aws-protocol-tests/model/restXml/http-payload.smithy#L390), so leaving in the simple tags regex

Reply:
We should keep Strings as is, with the tags if present